test/test_integration_check.py

```diff
@@ -0,0 +1,192 @@
+            self.assertIn("Warning: Running the command with '--insecure' makes your system vulnerable to security attacks",
+                          str(mock_log.output))
+
+    def test_check_specified_role_get_insecure_prints_warning_in_log(self):
+        with self.assertLogs('', level='DEBUG') as mock_log:
+            self.invoke_cli(self.cli_auth_params,
+                                     ['--insecure', 'check', '-i', 'dev:variable:somevariable', '-p', 'read', '-r', 'dev:user:permitted_user'])
+            self.assertIn("Warning: Running the command with '--insecure' makes your system vulnerable to security attacks",
+                          str(mock_log.output))
+
+    @integration_test()
+    def test_check_without_id_returns_help(self):
+        with redirect_stderr(self.capture_stream):
+            self.invoke_cli(self.cli_auth_params,
+                            ['check'], exit_code=1)
+        self.assertIn("Error the following arguments are required:", self.capture_stream.getvalue())
+
+    @integration_test(True)
+    def test_check_short_with_help_returns_check_help(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-h'])
+        self.assertIn("Name:\n  check", output)
+
+    @integration_test(True)
+    def test_check_long_with_help_returns_check_help(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '--help'])
+        self.assertIn("Name:\n  check", output)
+
+    '''
+    Tests to check for a privilege on a resource with a default role
+    '''
+    @integration_test(True)
+    def test_check_existent_privilege_on_permitted_resource_with_default_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:variable:somevariable', '-p', 'read'])
+        self.assertIn(output, "true\n")
+
+    @integration_test(True)
+    def test_check_non_existent_privilege_on_permitted_resource_with_default_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:variable:somevariable', '-p', 'execute'])
+        self.assertIn(output, ["true\n", "false\n"])
+
+    @integration_test(True)
+    def test_check_existent_privilege_on_non_permitted_resource_with_default_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:webservice:someservice', '-p', 'read'])
+        self.assertIn(output, "false\n")
+
+    @integration_test(True)
+    def test_check_non_existent_privilege_on_non_permitted_resource_with_default_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:policy:somepolicy', '-p', 'execute'])
+        self.assertIn(output, ["true\n", "false\n"])
```

Comment:
Does the output of these tests change? I'm curious how `true` and `false` can both be acceptable responses when nothing is changing on the policy.

Reply:
i'm currently trying to experiment this. i'm having difficulties as the Conjur instance i'm running on has super admin powers, which renders any check on an existent resource to be true. here, the expected output should be returning false, that the present role (assuming not admin) does not have access to the privilege but being admin in my testing, i'm always receiving true

Reply:
resolved